main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -1108,6 +1108,82 @@ object ResolutionError {
     override def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error raised to indicate a struct does not exist in a `new Struct { ... } @ r` expression
+    *
+    * @param struct the names of the struct
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStruct(struct: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Nonexistent struct type"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s""">> A struct type with the name `${struct}` does not exist
+         |
+         |${code(loc, "nonexistent struct")}
+         |""".stripMargin
+    }
+  }
+
+  /**
+    * An error raised to indicate a struct is missing a required field in `struct.field` or `struct.field = value` expression
+    *
+    * @param field the names of the missing fields
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStructField(struct: String, field: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Struct $struct does not have the field $field"
```

Comment:
Undefined field foo on struct bar

We prefer the broken thing to come first. Compare:

Undefined struct field 'Agee' on Agee
to
Struct Animal does not have field Agee